packages/arb-ts/src/lib/bridge_helpers.ts

```diff
@@ -265,14 +307,14 @@ export class BridgeHelper {
       ),
     }
 
-    const logs = l2Transaction.logs.filter(
+    const logs = l1Transaction.logs.filter(
       log =>
         log.topics[0] === eventTopics.InboxMessageDelivered ||
         log.topics[0] === eventTopics.InboxMessageDeliveredFromOrigin
     )
 
     if (logs.length === 0) return undefined
-    return logs.map(log => BigNumber.from(log.topics[1]))
+    return logs.map(log => BigNumber.from(log.topics[1]))[0]
```

Comment:
I don't think we should return the first element of the array. We can keep this neutral since you could have triggered the event multiple times.

Reply:
fair